server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1249,4 +1263,115 @@ public void testPrimaryReceivesDocsDuringReplicaRecovery() throws Exception {
         waitForSearchableDocs(2, nodes);
     }
 
+    /**
+     * This test validates that segment replications works with random compression mode
+     * @throws Exception
+     */
+    public void testWithDifferentCompressionModes() throws Exception {
+        final String nodeA = internalCluster().startNode();
+        final String nodeB = internalCluster().startNode();
+        final Settings settings = Settings.builder()
+            .put(indexSettings())
+            .put(
+                EngineConfig.INDEX_CODEC_SETTING.getKey(),
+                randomFrom(CodecService.DEFAULT_CODEC, CodecService.BEST_COMPRESSION_CODEC, CodecService.LUCENE_DEFAULT_CODEC)
+            )
+            .build();
+        createIndex(INDEX_NAME, settings);
+        final int initialDocCount = scaledRandomIntBetween(100, 200);
+        try (
+            BackgroundIndexer indexer = new BackgroundIndexer(
+                INDEX_NAME,
+                "_doc",
+                client(),
+                -1,
+                RandomizedTest.scaledRandomIntBetween(2, 5),
+                false,
+                random()
+            )
+        ) {
+            indexer.start(initialDocCount);
+            waitForDocs(initialDocCount, indexer);
+            refresh(INDEX_NAME);
+            waitForSearchableDocs(initialDocCount, nodeA, nodeB);
+
+            final int additionalDocCount = scaledRandomIntBetween(0, 200);
+            final int expectedHitCount = initialDocCount + additionalDocCount;
+            indexer.start(additionalDocCount);
+            waitForDocs(expectedHitCount, indexer);
+
+            flushAndRefresh(INDEX_NAME);
+            waitForSearchableDocs(expectedHitCount, nodeA, nodeB);
+
+            ensureGreen(INDEX_NAME);
+            verifyStoreContent();
+        }
+    }
+
+    /**
+     * This test validates that segment replications works for older codec implementations
+     * @throws Exception
+     */
+    public void testWithOlderCodecs() throws Exception {
```

Comment:
I'm not sure this test is giving us much when outside of an upgrade scenario.  With pri & replica running the same version of Lucene, this is only testing the compatibility of the codec selected with a reader on the same lucene version.

Reply:
I think this test adds value as
1. It catches the string [matching checks](https://github.com/dreamer-89/OpenSearch/commit/c0be2bc0f1a5f961ca7fa16f8e5c932adece557c) bug which previously escaped. It basically covers/mimics use-cases where non-default codecs are used (e.g. kNN). Having this test in place, will prevent similar issues in future. 
2. This test ensures segrep doesn't break for non-default codecs. 
    a). With some of our recent developments around codec changes (e.g. support [rolling upgrades](https://github.com/opensearch-project/OpenSearch/pull/7698) where we add validations on checkpoint processing, version to codec name mapping, this test will be super useful.
   b). Codec names are used in primary <-> replica communication (via ReplicationCheckpoint) and in essence, it is center piece of information. Having a test with non-default codec, may catch issues which doesn't appear for default ones.

Reply:
Discussed with @mch2 offline, reading lucene jar to figure out codecs can break in future with directory changes. Removing this test for now as bwc tests with segrep enabled provide same coverage.